### PR TITLE
feat: Added the Permission Required Section

### LINKED ISSUE #1516: Update README of permissions needed in the app and reason
**Actual Behaviour**
Currently the README is not showing what permissions are needed and explaining it

**Expected Behaviour**

Permissions required and of explanation why it is needed in the README.

### PR DESCRIPTION
**Pull request by Sjsingh101**
fixes #1516

I have added the permissions required section which help the other developers who are beginning to read this project and give a brief idea why it is necessary to add these permissions

image

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x ] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +128,14 @@ To debug over Wi-Fi: http://blog.fossasia.org/android-app-debugging-over-wifi-fo
 Note :
 1. If you built your own hardware, change VendorID and/or ProductID in [CommunicationHandler.java](https://github.com/fossasia/pslab-android/blob/master/app/src/main/java/org/fossasia/pslab/communication/CommunicationHandler.java)
 
+### Permissions Required
+
+1. RECORD_AUDIO : It is required because in oscilloscope we are accepting inputs from the mic so we need to add this permission in order to contiune the proper functioning of this feature.You can find its implementation in [AudioJack.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/AudioJack.java).
+2. ACCESS_FINE_LOCATION And INTERNET:  They are required because in compass and Map we need to get the location access so that 	the full use of these features can be taken.You can find its implementation in [GPSLogger.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/GPSLogger.java) .
```
[the remark this answers is not in the document]
Comment (Sjsingh101) [reply README.md]: extra space is removed and capitalization error is also removed 
@CloudyPadmal please review the pull request

File: README.md [hunk 2]
```diff
@@ -128,6 +128,13 @@ To debug over Wi-Fi: http://blog.fossasia.org/android-app-debugging-over-wifi-fo
 Note :
 1. If you built your own hardware, change VendorID and/or ProductID in [CommunicationHandler.java](https://github.com/fossasia/pslab-android/blob/master/app/src/main/java/org/fossasia/pslab/communication/CommunicationHandler.java)
 
+### Permissions Required
+
+1. Record_Audio : It is required because in oscilloscope we are accepting inputs from the mic so we need to add this permission in order to contiune the proper functioning of this feature.You can find its implementation in [AudioJack.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/AudioJack.java).
```
Comment (CloudyPadmal) [inline README.md]: 1. contiune --> continue
2. feature.You --> feature. You

File: README.md [hunk 3]
```diff
@@ -128,6 +128,13 @@ To debug over Wi-Fi: http://blog.fossasia.org/android-app-debugging-over-wifi-fo
...
+2. Access_Fine_Location And Internet : They are required because in compass and Map we need to get the location access so that the full use of these features can be taken.You can find its implementation in [GPSLogger.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/GPSLogger.java).
```
Comment (CloudyPadmal) [inline README.md]: 1. Access_Fine_Location And Internet --> Access_Fine_Location and Internet
2. taken.You --> taken. You

File: README.md [hunk 4]
```diff
@@ -128,6 +128,13 @@ To debug over Wi-Fi: http://blog.fossasia.org/android-app-debugging-over-wifi-fo
...
+3. Write_External_Storage : It is required for writing log files in the storage as the test results are stored in it for future analysis.
+4. Read_External_Storage : While writing logs in the storage,[CSVLogger.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/CSVLogger.java) first checks whether there is any CSVLogger directory exist or not and that require this read permission and its another use comes while implementing LeakCanary as it require read storage permission to detect the memory leaks in application.
```
Comment (CloudyPadmal) [inline README.md]: 1. storage,[CSVLogger --> storage, [CSVLogger

File: README.md (hunk 2 above)
Comment (cweitat) [reply README.md]: Keep it simple, don't explain for too long.
For oscilloscope to accept inputs from the phone microphone. You can find.....

File: README.md (hunk 3 above)
Comment (cweitat) [reply README.md]: For use in lux meter and compass to get the coordinates for tagging the data on the map. You can find ....

File: README.md (hunk 4 above)
Comment (cweitat) [reply README.md]: Point 3:
For storing log files from instruments that can be transferred out for future analysis.

**Comment by Sjsingh101** [comment]
@CloudyPadmal i have done the all purposed changes  by @cweitat

File: README.md [hunk 5]
```diff
@@ -128,6 +128,13 @@ To debug over Wi-Fi: http://blog.fossasia.org/android-app-debugging-over-wifi-fo
...
 
+### Permissions Required
+
+1. Record_Audio : It is required for oscilloscope to accept inputs from the phone inbuilt microphone. You can find its implementation in [AudioJack.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/AudioJack.java).
+2. Access_Fine_Location and Internet : It is required for use in lux meter and compass to get the coordinates for tagging the data on the map. You can find its implementation in [GPSLogger.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/GPSLogger.java).
+3. Write_External_Storage : It is required for storing log files from instruments that can be transferred out for future analysis.
+4. Read_External_Storage : While writing logs in the storage, [CSVLogger.java](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/others/CSVLogger.java) first checks whether there is any CSVLogger directory exist or not and that require this read permission and its another use comes while implementing LeakCanary as it require read storage permission to detect the memory leaks in application.
```
[the remark this answers is not in the document]
Comment (Sjsingh101) [reply README.md]: @CloudyPadmal  i have done the changes

**Comment by CloudyPadmal** [comment]
Hi, please update the working branch.

**Comment by Sjsingh101** [comment]
@sk9331657  please 

> @Sjsingh101 Squash Your Commits.

done

**Comment by harsh-2711** [comment]
@Sjsingh101 Can you please squash and rebase the branch. Else the PR looks good.

**Pull request merged**